spanner/api/Spanner.Samples/WriteWithTransactionUsingDmlCoreAsync.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_dml_getting_started_update]
+
+using Google.Cloud.Spanner.Data;
+using System;
+using System.Threading.Tasks;
+
+public class WriteWithTransactionUsingDmlCoreAsyncSample
+{
+    public async Task<int> WriteWithTransactionUsingDmlCoreAsync(string projectId, string instanceId, string databaseId)
+    {
+        // This sample transfers 200,000 from the MarketingBudget
+        // field of the second Album to the first Album. Make sure to run
+        // the addColumn and writeDataToNewColumn samples first,
```

Comment:
Mostly for @skuruppu: are `addColumn` and `writeDataToNewColumn` good names for directing users to the other samples?

Reply:
That's a very good point. `addColumn` and `writeDataToNewColumn` is if you're using the command line tool. Since that's going away, we should refer to these with the sample names.

```suggestion
        // the AddColumnAsyncSample and WriteDataToNewColumnAsyncSample first,
```

@laljikanjareeya I can't find the `spanner_update_data` sample in the rewrite. Would you be able to add it to this PR?